### ISSUE TITLE
Trying to reproduce MJEPA-L on Audioset

### ISSUE THREAD

**Issue by peterwisu** [facebookresearch/MJEPA#4]
Hi,

I'm trying to reproduce the base ViT-L results, and I wanted to ask about something I noticed during pre-training.

I followed the released config for the base ViT-L and trained for 600 epochs, with the same total batch size of 512:
https://github.com/facebookresearch/MJEPA/blob/main/configs/train/vitl16/pretrain-224px-16f.yaml

Near the end of training there is a jump in the loss. The paper reports 150k pre-training steps, which is epoch 500 (500 epochs × 300 ipe) and before this jump, so I used the epoch-500 checkpoint for evaluation. On the AudioSet-20K audio-video frozen evaluation I got around 18.7 mAP, compared to 42.90 in the paper.

Looking at the individual loss terms (plot attached below), the intra-modal losses (a2a, v2v, av2av) reach their lowest point around epoch 100 and then go back up before levelling off, while all the cross-modal losses (a2v, v2a, av2a, av2v, a2av, v2av) keep decreasing for the rest of training. This makes me suspect the representation may be collapsing after epoch 100. I also evaluated the epoch-100 checkpoint and got around 22 mAP, which is higher than epoch 500.

<img width="1950" height="1740" alt="Image" src="https://github.com/user-attachments/assets/47f42124-7e7b-43ed-85d1-957a1828fed6" />

Can you heIp me resolve this and also I have a few questions
1. Did you observe similar behaviour in the individual losses during pre-training? If so, do you have any suggestions on how to avoid it, so the results are closer to the paper?  
2.  The config is set to 600 epochs (180k steps), but the paper reports 150k steps (epoch 500). Did you train for 600 epochs and use the epoch-500 checkpoint, or set training to 500 epochs and use the final checkpoint? If you trained for 600 epochs, did you also see the jump in the loss near the end?

3. Is there anything in my setup that I might have missed or done differently from your training?

Thank you

**Comment by revantteotia** [comment]
Typical ema based jepa loss curve behaves like this: https://github.com/facebookresearch/vjepa2/issues/56. 
It first drops quickly, then rises and then plateaus (with a small rise). They do not keep on decreasing. 
We observed something like this in our training runs (showing total loss): 
<img width="300" height="300" alt="Image" src="https://github.com/user-attachments/assets/f6c4e6eb-7725-49a8-bf71-85f9ef11c305" />

By looking at your curves, it seems like the representations are collapsing slowly. You might need to adjust hyperparameters based on your training dataset. 

- It might be training data mismatch? Are you training your model on full Audioset 2M? is data pre-processing matching the paper? (audio sample rate? video frame rate?) 

- We trained for full 600 epochs. But observed that after 120k-150k steps, the downstream performance became saturated and do not rise fast. So I picked 150k steps checkpoints for my training runs. I do not remember seeing a loss jump in the end. I think, in your runs we see a jump because the representations are probably collapsed. 

You might need to recheck the training data (and pre processing) and adjust your hyperparameters accordingly. 

Good luck :)

**Comment by peterwisu** [comment]
Hi thank you for your reply,

1. Yes I am using audioset 2M as well the data preprocess both video fps and audio sample rate  kept the config file the same as given in the github.
2. Yes i agree i think the jump is causing from the collpase and i think it loss that casue the collapse is the cross-modal loss after epoch 100 since it keep decreasing while intra modal loss is increase after the drop. Do you thingk adjusting these  coefficent for each loss would solve this collapse problem? 
3. And also in the video preprocessing  do you resize all the video to 256px (height side) first and then in data loader you crop and resize again to 224px in ViT-L

**Comment by revantteotia** [comment]
1. If the training data and pre-processing is same then it should train well. I trained using the same config on two different HPCs and the training/eval runs were pretty consistent across clusters (although I used the same dataset copy). 

2. In all of my runs, I kept the coefficient same. Although you can try adjusting them to keep the training stable.

3.  Yes correct. We resized the videos during pre-processing. As far as I remember, we resized shorter side in each video to 256px (or 320px, I don't remember exactly - I don't have access anymore to confirm) while preserving aspect ratio. It also improves dataloading speed.
